python/ray/tune/integration/mlflow.py

```diff
@@ -0,0 +1,337 @@
+    use the mlflow API inside the your training function and it will
+    automatically get reported to the correct run.
+
+    For basic usage, just prepend your training function with the
+    ``@mlflow_mixin`` decorator:
+
+    .. code-block:: python
+
+        from ray.tune.integration.mlflow import mlflow_mixin
+
+        @mlflow_mixin
+        def train_fn(config):
+            ...
+            mlflow.log_metric(...)
+
+    You can also use Mlflow's autologging feature if using a training
+    framework like Pytorch Lightning, XGBoost, etc. More information can be
+    found here (https://mlflow.org/docs/latest/tracking.html#automatic
+    -logging).
+
+    .. code-block:: python
+
+        from ray.tune.integration.mlflow import mlflow_mixin
+
+        @mlflow_mixin
+        def train_fn(config):
+            mlflow.autolog()
+            xgboost_results = xgb.train(config, ...)
+
+    The Mlflow configuration is done by passing a ``mlflow`` key to
+    the ``config`` parameter of ``tune.run()`` (see example below).
+
+    The content of the ``mlflow`` config entry is used to
+    configure Mlflow. Here are the keys you can pass in to this config entry:
+
+    Args:
+        tracking_uri (str): The tracking URI for MLflow tracking. If using
+            Tune in a multi-node setting, make sure to use a remote server for
+            tracking.
+        experiment_id (str): The id of an already created MLflow experiment.
+            All logs from all trials in ``tune.run`` will be reported to this
+            experiment. If this is not provided, you must provide an
+            ``experiment_name``.
+        experiment_name (str): The name of an already created MLflow
+            experiment. All logs from all trials in ``tune.run`` will be
+            reported to this experiment. If this is not provided, you must
+            provide an ``experiment_id``.
+
+    Example:
+
+    .. code-block:: python
+
+        from ray import tune
+        from ray.tune.integration.mlflow import mlflow_mixin
+
+        import mlflow
+
+        # Create the Mlflow expriment.
+        mlflow.create_experiment("my_experiment")
+
+        @mlflow_mixin
+        def train_fn(config):
+            for i in range(10):
+                loss = self.config["a"] + self.config["b"]
+                mlflow.log_metric(key="loss", value=loss})
+            tune.report(loss=loss, done=True)
+
+        tune.run(
+            train_fn,
+            config={
+                # define search space here
+                "a": tune.choice([1, 2, 3]),
+                "b": tune.choice([4, 5, 6]),
+                # mlflow configuration
+                "mlflow": {
+                    "experiment_name": "my_experiment",
+                    "tracking_uri": mlflow.get_tracking_uri()
+                }
+            })
+    """
+    if mlflow is None:
+        raise RuntimeError("mlflow has not been installed. Please `pip "
```

Comment:
I think we should decide for one way to spell MLFlow - I'd propose using MLFlow as the proper noun in documentation everywhere

Reply:
Yes, good point. This should all be fixed now!